lib/blob/blobstore.c

```diff
@@ -2777,6 +2777,42 @@ blob_calculate_lba_and_lba_count(struct spdk_blob *blob, uint64_t io_unit, uint6
 	}
 }
 
+static inline bool
+blob_ancestor_calc_lba_and_lba_count(struct spdk_blob *blob, uint64_t io_unit, uint64_t length,
+				     uint64_t *lba, uint64_t *lba_count)
+{
+	*lba_count = length;
+	struct spdk_blob *orig_blob = blob;
+	if (!bs_io_unit_is_allocated(orig_blob, io_unit)) {
+		if (orig_blob->parent_id == SPDK_BLOBID_INVALID) {
+			goto error;
+		} else {
+			while (blob->parent_id != SPDK_BLOBID_INVALID) {
+				spdk_blob_id blob_id = blob->parent_id;
+				blob = blob_lookup(blob->bs, blob_id);
+				if (blob == NULL) {
+					return false;
+				} else {
+					if (!bs_io_unit_is_allocated(blob, io_unit)) {
+						continue;
+					}
+					*lba = bs_blob_io_unit_to_lba(blob, io_unit);
+					return true;
+				}
+			}
+			goto error;
+		}
+	} else {
+		*lba = bs_blob_io_unit_to_lba(blob, io_unit);
+		return true;
+	}
+error:
+	assert(orig_blob->back_bs_dev != NULL);
+	*lba = bs_io_unit_to_back_dev_lba(orig_blob, io_unit);
```

Comment:
do we need to set these in case of error?

Reply:
It is kept same existing funtion`blob_calculate_lba_and_lba_count`  behaviour where the default lba is calculated in fail scenario.